### ISSUE TITLE
x-ms-error-response support

### ISSUE THREAD

**Issue by RikkiGibson** [Azure/autorest.typescript#283]
The AutoRest test server has a fixture called [xms-error-response.json](https://github.com/Azure/autorest.testserver/blob/b6e49c35fb70c82ae1837507e069aca81d89a092/swagger/xms-error-responses.json). See the proposal [here](https://github.com/Azure/autorest/blob/bc0dcde0a05606a1b102c0e08c1798fa4da4dcc8/docs/proposals/generator-specific-settings/improved-exceptions.md#proposed-solutions).

I started this work when trying to get full coverage but realized that some design work will need to happen first.

We should figure out what support for this extension looks like in TypeScript. Note that I don't see usage of this at this time in azure-rest-api-specs, so let's consider this **low priority**.
